src/lib/dataviews/DataViewWrapperBase.ts

```diff
@@ -0,0 +1,96 @@
+import { WithEvents } from '@/core/mixins/WithEvents';
+import { Layer, Source } from '@/viz';
+import { Filter } from '@/viz/filters/types';
+import { AggregationType } from '@/maps/MapsDataviews';
+import { DataViewImplBase } from './DataViewImplBase';
+import { DataViewModeBase, DataViewModeAlias } from './mode/DataViewModeBase';
+
+const OPTION_CHANGED_DELAY = 100;
+
+export abstract class DataViewWrapperBase extends WithEvents {
+  protected dataviewImpl!: DataViewImplBase<DataViewModeBase>;
+
+  /**
+   * optionChanged timeout to prevent multiple
+   * calls when user sets several options in a row
+   */
+  private optionChangedTimeoutId?: number;
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  constructor(dataSource: Layer | Source, column: string, options: any) {
+    super();
+
+    const mode = options.mode || DataViewModeAlias.VIEWPORT;
+    this.buildImpl(dataSource, column, options, mode);
+
+    // bind events with the mode
+    this.bindEvents();
+
+    this.on('optionChanged', () => {
+      // timeout prevents multiple calls to getData
+      // when user sets several options in a row
+      if (this.optionChangedTimeoutId) {
+        window.clearTimeout(this.optionChangedTimeoutId);
+      }
+
+      this.optionChangedTimeoutId = window.setTimeout(
+        () => this.emit('dataUpdate'),
+        OPTION_CHANGED_DELAY
+      );
+    });
+  }
+
+  public getData() {
+    return this.dataviewImpl.getData();
+  }
+
+  public addFilter(filterId: string, filter: Filter) {
+    this.dataviewImpl.addFilter(filterId, filter);
+  }
+
+  public removeFilter(filterId: string) {
+    this.dataviewImpl.removeFilter(filterId);
+  }
+
+  public get column() {
+    return this.dataviewImpl.column;
+  }
+
+  public set column(column: string) {
+    this.dataviewImpl.column = column;
+    this.emit('optionChanged');
```

Comment:
what's the purpouse of 'optionChanged'? who should listen and react to it?

Reply:
I added it to listen when some attribute of the Dataviews has changed in order to fetch the new data and then fire the event `dataUpdated` with the new data retrieved by Maps API. Finally I removed that behaviour and let users of the dataView to fetch the data by themselves so I think I should use `dataUpdated` instead of `optionChanged` and remove this.

By this way the dataViews will fire the event `dataUpdated` when some option has changed in the same way DataView local fires this event when the viewport has changed.

Reply:
However, with `optionChanged` we were managing through a timeout when to fire dataUpdated in order to prevent fires this event multiples times if the options of the dataView has changed in a row. This is, if we set three options of our dataView:

```javascript
categoryDataView.column = column;
categoryDataView.operationColumn = operationColumn;
categoryDataView.operation = operation;
```

With `optionChanged` + `timeout` we are firing the event `dataUpdate` just one time whereas if we remove the `optionChanged` and use `dataUpdate` instead in the example above this event would be firing three times in a row.

Reply:
If that event is just used internally for debouncing, to trigger less requests, it's ok, but let's try to keep it internal. In fact, I would suggest also extracting the debounce to utils